frontend/src/__tests__/api.test.js

```diff
@@ -96,95 +95,230 @@ function sessionsTests(api = { apiGET, apiPOST }) {
         // make sure the id of our new session came back
         expect(withNewSessions.map(x => x.id)).toContain(createdSession.id);
 
-        // now delete the session and make sure it's gone
-        const resp2 = await apiPOST("/sessions/delete", {
-            id: createdSession.id
+        // save this session for use in later tests
+        session = resp1.payload;
+    });
+
+    it("fetch sessions", async () => {
+        const resp = await apiGET("/sessions");
+
+        expect(resp).toMatchObject({ status: "success" });
+        checkPropTypes(PropTypes.arrayOf(sessionPropTypes), resp.payload);
+    });
+
+    it("update a session", async () => {
+        const id = session.id;
+        const newData = { id, rate2: 57.75 };
+        const resp1 = await apiPOST("/sessions", newData);
+        expect(resp1).toMatchObject({ status: "success" });
+        expect(resp1.payload).toMatchObject(newData);
+
+        // get the sessions list and make sure we're updated there as well
+        const resp2 = await apiGET("/sessions");
+        // filter session list to get the updated session obj
+        const updatedSession = resp2.payload.filter(s => s.id == id);
+        expect(updatedSession).toContainObject(newData);
+    });
+
+    it("throw error when `name` is empty", async () => {
+        // create new session with empty name
+        const newData = { name: "" };
+        const resp1 = await apiPOST("/sessions", newData);
+        // expected an error
+        expect(resp1).toMatchObject({ status: "error" });
+        checkPropTypes(errorPropTypes, resp1);
+    });
+
+    it("throw error when `name` is not unique", async () => {
+        // name identical to the exisiting session
+        const newData = { name: session.name };
+        // POST to create new session
+        const resp1 = await apiPOST("/sessions", newData);
+
+        // expected an error as name not unique
+        expect(resp1).toMatchObject({ status: "error" });
+        checkPropTypes(errorPropTypes, resp1);
+    });
+
+    it("throw error when deleting item with invalid id", async () => {
+        const resp1 = await apiPOST("/sessions/delete", {
+            id: session.id + Math.floor(Math.random() * 100) + 1 // returns a random integer from 1 to 100
```

Comment:
I don't like this....get all the sessions, find the largest ID and then add 1 to it to ensure you found an ID that doesn't exist. Also, try to delete something with `id=null`